opt4j-optimizers/src/main/java/org/opt4j/optimizers/ea/NonDominatedFronts.java

```diff
@@ -0,0 +1,192 @@
+package org.opt4j.optimizers.ea;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.opt4j.core.Individual;
+import org.opt4j.core.Objective;
+import org.opt4j.core.Objectives;
+
+/**
+ * The {@link NonDominatedFronts} sorts each evaluated {@link Individual} into
+ * fronts based on the number of other individuals it is dominated by. The first
+ * front consists of points that are not dominated at all and so on.
+ * 
+ * @author Fedor Smirnov
+ *
+ */
+public class NonDominatedFronts extends ArrayList<List<Individual>> {
+
+	private static final long serialVersionUID = -7008617060878292974L;
+
+	/**
+	 * Creates the {@link NonDominatedFronts} for the given collection of
+	 * {@link Individual}s.
+	 * 
+	 * @param individuals
+	 *            the {@link Individual}s that are sorted into non dominated fronts
+	 */
+	public NonDominatedFronts(Collection<Individual> individuals) {
+		generateFronts(individuals);
+	}
+
+	/**
+	 * Sorts the given {@link Individual}s into non-dominated fronts.
+	 * 
+	 * @param individuals
+	 *            the collection of {@link Individual}s that shall be sorted
+	 */
+	public void generateFronts(Collection<Individual> individuals) {
```

Comment:
when should I use the constructor, when the ```generateFronts()``` ? Both are public. Same for ```#add(List<Indivudual> i)```.

Reply:
You are right; generateFronts() should not be public